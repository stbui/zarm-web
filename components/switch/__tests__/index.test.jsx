import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Switch from '../index';

describe('Switch', () => {
  it('renders normal Switch correctly', () => {
    const wrapper = render(
      <div>
        <Switch />
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Switch correctly with more props', () => {
    const wrapper = render(
      <div>
        <Switch disabled size="sm" />
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when receving new value', () => {
    const wrapper = mount(
      <Switch defaultChecked={false} />,
    );

    expect(wrapper.find('.zw-switch').hasClass('zw-switch--checked')).toBeFalsy();
    wrapper.find('.zw-switch').simulate('click');
    expect(wrapper.find('.zw-switch').hasClass('zw-switch--checked')).toBeTruthy();
  });

  it('behaves correctly when toggling status', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Switch defaultChecked={false} onChange={onChange} />,
    );

    const wrapperOpen = mount(
      <Switch defaultChecked={false} onChange={onChange} />,
    );

    wrapper.find('.zw-switch').simulate('click');
    expect(onChange).toHaveBeenCalledWith(true);

    wrapperOpen.find('.zw-switch').simulate('click');
    expect(onChange).toHaveBeenCalledWith(true);
  });

  it('switch with loading status', () => {
    const wrapper = mount(
      <Switch defaultChecked />,
    );
    expect(wrapper.exists('.zw-switch--loading')).toEqual(false);
    wrapper.setProps({ loading: true });
    expect(wrapper.exists('.zw-switch--loading')).toEqual(true);
  });
});
