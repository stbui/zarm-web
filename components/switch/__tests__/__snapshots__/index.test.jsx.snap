// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders Switch correctly with more props 1`] = `
<div>
  <button
    aria-checked="false"
    class="zw-switch zw-switch--disabled zw-switch--sm"
    role="switch"
    type="button"
  />
</div>
`;

exports[`Switch renders normal Switch correctly 1`] = `
<div>
  <button
    aria-checked="false"
    class="zw-switch"
    role="switch"
    type="button"
  />
</div>
`;
